crypto/aes_gcm.go

```diff
@@ -0,0 +1,167 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"hash"
+
+	"github.com/bndw/pick/errors"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+type AESGCMClient struct {
```

Comment:
Yup. Will update.
IMO it now makes sense to rename 'AESGCMConfig' to just 'aesGcmConfig' and not uppercase everything.
What do you think?

I will push all other changes in a few hours, then squash everything. Hope we can merge soon.